### PR TITLE
CSS Frameworks CA

### PR DESCRIPTION
**Pull request by kyrregjerstad**
# Summary
This PR sets up the project with Vite and TypeScript, adds Bootstrap and Sass, and includes basic web components.

I use [pnpm](https://pnpm.io) as a package manager. 

## Changes

- [x] **Build Tool:** Adopted Vite with TypeScript support.
- [x] **UI Library:** Installed Bootstrap and Popper.js.
- [x] **Styling:** Integrated Sass.
- [x] **Web Components:** Created basic components for different UI elements.
- [x] **Bootstrap Customization:** Overrode default Bootstrap styles.
- [x] **Documentation:** TODO: create readme

### CODE REVIEW

File: src/scss/styles.scss [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+    overflow: hidden;
+    border-radius: 8px;
+    max-height: 80vh;
+  }
+
+  .post-content {
+    margin: 16px 0;
```
Comment (Fermain) [inline src/scss/styles.scss]: `.my-3`

File: src/scss/styles.scss [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+      .timestamp {
+        color: gray;
+        font-size: 12px;
+      }
+    }
+  }
+
+  .post-image {
+    width: 100%;
```
Comment (Fermain) [inline src/scss/styles.scss]: `.w-100`

File: src/scss/styles.scss [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    height: 100%;
```
Comment (Fermain) [inline src/scss/styles.scss]: `.h-100`

File: src/scss/styles.scss [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    object-fit: cover;
```
Comment (Fermain) [inline src/scss/styles.scss]: `.object-fit-cover`

File: src/scss/styles.scss [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    overflow: hidden;
```
Comment (Fermain) [inline src/scss/styles.scss]: `.overflow-hidden`

File: src/scss/styles.scss [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    border-radius: 8px;
```
Comment (Fermain) [inline src/scss/styles.scss]: `.rounded-2`

File: src/style.css [hunk 7]
```diff
@@ -0,0 +1,97 @@
+:root {
```
Comment (Fermain) [inline src/style.css]: There should not be a src/style.css file - is this a mistake?

**Review by Fermain — CHANGES_REQUESTED** [review]
This is absurdly overegineered considering what we asked for, but in the best possible way. I love your approach to TS custom elements, really creative compromise to not using a JS framework.

In order to make the best possible CA, your mission is to remove as many custom styles as you can from your SCSS. Either update your HTML to use more of the provided utility classes, or use `@extend` within SCSS to proxy your custom classes to existing bootstrap classes. There should be almost no custom styles at all in your file, instead variable customisation is the key.

Not sure what the css file inside src is for, but that should be moved or deleted.

File: src/style.css (hunk 7 above)
Comment (kyrregjerstad) [reply src/style.css]: ah, leftover from the initial Vite starter

**Pull request merged**